patches/hideSubjectsWithNoGrades/script.js

```diff
@@ -0,0 +1,34 @@
+const SEMESTERS = 2;
+const renderVisibilityButtons = () => {
+    for (let i = 0; i < SEMESTERS; i++) {
+        hideEmptyFinalGradesInfo(i)
+    }
+}
+
+const getSemestersContainer = () => document.querySelector("section > section > .mobile__frame > .content-container").children
+
+const hideEmptyFinalGradesInfo = async (i) => {
+    await window.waitForRender(() => getSemestersContainer()[i].querySelector("article"), document.querySelector("section > section > .mobile__frame > .content-container"))
+    const container = getSemestersContainer()[i]
+    const subjects = container.querySelectorAll("article")
+    for (const subject of subjects) {
+        const finalGrades = document.querySelector(".tile__content:last-of-type")
+        const predictedFinalGrade  = finalGrades.children[0]
+        const finalGrade = finalGrades.children[1]
+        if (!predictedFinalGrade.querySelector(".info-text > span")?.innerText?.trim()) {
```

Comment:
![image](https://github.com/user-attachments/assets/3c8dead1-5882-47fe-a714-f4322c234197)

```suggestion
        if (!predictedFinalGrade?.querySelector(".info-text > span")?.innerText?.trim()) {
```